database/buildSchema/43_views_functions_triggers.sql

```diff
@@ -1542,12 +1542,332 @@ $$
 LANGUAGE sql
 STABLE;
 
+-- Does the "heavy lifting" to deduce the assigners/reviewers list for a single
+-- application as well as the available actions for a particular
+-- reviewer/assigner for that application. Used to populate the
+-- `application_reviewer_action` table and the reviewer/assigner lists on each
+-- application whenever an event occurs that would change these values
+CREATE OR REPLACE FUNCTION single_application_detail (applicationId int, userId int DEFAULT 0)
```

Comment:
This function is a pure copy/paste from the existing list function. Differences are
- only for a single application (passed in `applicationId` parameter, and additional `WHERE` clause).
- removed all columns other than these 4 complicated ones -- the others are all just joined in the list query as before.

Reply:
Still has a `DEFAULT` value since when generating the assigners and reviewers lists, we don't need to specify a user.

Reply:
Not quite sure what you mean. It's only doing one thing I think -- gets the values for the reviewer/assigner lists. The list actions come from the get list function, which return the pre-computed values that this function calculated earlier.

Reply:
It seems to be called separately with userId = 0 for assigners and reviewers, i think below can be separated

```sql
            CASE WHEN count(assigner_user.id) = 0 THEN
                NULL
            ELSE
                ARRAY_AGG(DISTINCT (CONCAT(assigner_user.first_name, ' ', assigner_user.last_name)::varchar))
            END AS assigners,
            CASE WHEN count(reviewer_user.id) = 0 THEN
                NULL
            ELSE
                ARRAY_AGG(DISTINCT (CONCAT(reviewer_user.first_name, ' ', reviewer_user.last_name)::varchar))
            END AS reviewers,
```

And the everything below

```sql
 -- REVIEWER ACTIONS
            CASE WHEN COUNT(*) FILTER (WHERE action_review_status_history.status = 'CHANGES_REQUESTED') != 0 THEN

...
```

Reply:
Yeah, I was able to just copy and paste all this logic from the existing function, so I'm reluctant to start picking it apart, even a little bit at this point -- potentially very messy and time-consuming.